driver/normalizer/normalizer.go

```diff
@@ -15,6 +15,141 @@ var Normalize = Transformers([][]Transformer{
+}
+
+func (op opArrHasParams) Check(st *State, n nodes.Node) (bool, error) {
+	arr, ok := n.(nodes.Array)
+	if !ok && arr != nil {
+		return false, nil
+	}
+
+    res, err := op.origArr.Check(st, arr)
+    if err != nil {
+    	return false, err
+	}
+	return res, nil
+}
+
+func (op opArrHasParams) Construct(st *State, n nodes.Node) (nodes.Node, error) {
+	arr, err := op.origArr.Construct(st, n)
+	if err != nil || arr == nil {
+		return n, err
+	}
+
+	arr2, ok := arr.(nodes.Array)
+	if !ok && arr2 != nil {
+		return nil, ErrExpectedList.New(arr2)
+	}
+
+	retVal := false
+
+	for _, n := range arr2 {
+		nobj, ok := n.(nodes.Object)
+		if !ok {
+			return nil, ErrExpectedObject.New(nobj)
+		}
+
+		objType, ok := nobj["@type"].(nodes.String)
+		if !ok {
+			return nil, ErrExpectedValue.New(nobj["@type"])
+		}
+
+		if objType == "ParamsKeyword" {
+			retVal = true
+			break
+		}
+	}
+
+	return nodes.Bool(retVal), nil
+}
+
+func funcDefMap(typ string) Mapping {
+	return MapSemantic(typ, uast.FunctionGroup{}, MapObj(
+		Fields{
+			{Name: "Body", Op: Var("body")},
+			{Name: "Identifier", Op: Obj{
+				uast.KeyType: String("IdentifierToken"),
+				uast.KeyPos:      Var("id_fnc_pos"),
+				"LeadingTrivia":  Any(),
+				"TrailingTrivia": Any(),
+				"IsMissing": Bool(false),
+				"Text":      Var("name"),
+				"Value":     Var("name"),
+				"ValueText": Var("name"),
+			}},
+			{Name: "ParameterList", Op: Obj{
+				uast.KeyType: String("ParameterList"),
+				uast.KeyPos: Any(),
+				"OpenParenToken": Any(),
+				"CloseParenToken": Any(),
+				"IsMissing": Bool(false),
+				"IsStructuredTrivia": Bool(false),
+				"Parameters": Var("params"),
+			}},
+			{Name: "ReturnType", Optional: "optReturn", Op: Cases("cases_return",
+				// Type = PredefinedType
+				Obj{
+					uast.KeyType: String("PredefinedType"),
+					uast.KeyPos: Any(),
+					"IsMissing": Bool(false),
+					"IsStructuredTrivia": Bool(false),
+					"IsUnmanaged": Any(),
+					"IsVar": Any(),
+					"Keyword": Obj{
+						uast.KeyType: Any(),
+						uast.KeyPos: Any(),
+						"IsMissing": Bool(false),
+						"LeadingTrivia": Any(),
+						"TrailingTrivia": Any(),
+						"Text": Var("rettype_text"),
+						"Value": Var("rettype_text"),
+						"ValueText": Var("rettype_text"),
+					},
+				},
+				// Type = Identifier (User type)
+				Var("rettype_ident"),
+			)},
+		},
+
+		Obj{
+			"Nodes": Arr(
+				UASTType(uast.Alias{}, Obj{
+					"Name": UASTType(uast.Identifier{}, Obj{
+						uast.KeyPos: Var("id_fnc_pos"),
+						"Name": Var("name"),
+					}),
+					"Node": UASTType(uast.Function{}, Obj{
+						"Body": Var("body"),
+						"Type": UASTType(uast.FunctionType{}, Fields{
+							{Name: "Arguments", Op: Var("params")},
+							{Name: "Returns", Optional: "optReturn", Op: Arr(
+								Cases("cases_return",
+									// Type = PredefinedType
+									UASTType(uast.Argument{}, Obj{
+										"Type": UASTType(uast.Identifier{}, Obj{
+											"Name": Var("rettype_text"),
```

Comment:
Looking at this it seems like we can just convert all `PredefinedType` nodes to `uast:Identifier` and be done with it.